src/SearchEngine/MASA.BuildingBlocks.SearchEngine.AutoComplete/AudoCompleteDocument.cs

```diff
@@ -0,0 +1,31 @@
+namespace MASA.BuildingBlocks.SearchEngine.AutoComplete;
+public class AudoCompleteDocument<TValue>
+{
+    public string Id { get; set; }
+
+    public string Text { get; set; }
+
+    public TValue Value { get; set; }
+
+
+    public AudoCompleteDocument()
+    {
+    }
+
+    public AudoCompleteDocument(string text, TValue value) : this()
+    {
+        Text = text;
+        Value = value;
+        Id = GetId();
+    }
+
+    public AudoCompleteDocument(string id, string text, TValue value) : this(text, value)
+    {
+        Id = id;
+    }
+
+    protected virtual string GetId()
+    {
```

Comment:
If set the Id, you will show `$"[{Value}]{Text}"`?

Reply:
If id is specified alone then it will be the value you specified, otherwise it is $"[{Value}]{Text}"